### PR TITLE
Parallel dequeue

### PR DESCRIPTION
**Pull request by SergeyGaluzo**
Addresses https://microsofthealth.visualstudio.com/Health/_workitems/edit/99473/

### CODE REVIEW

File: src/Microsoft.Health.TaskManagement/JobHosting.cs [hunk 1]
```diff
@@ -37,46 +37,50 @@ public JobHosting(IQueueClient queueClient, IJobFactory jobFactory, ILogger<JobH
...
+                        }
+
+                        if (nextJob != null)
+                        {
+                            await ExecuteJobAsync(nextJob, useHeavyHeartbeats);
+                        }
+                        else
+                        {
+                            await Task.Delay(TimeSpan.FromSeconds(PollingFrequencyInSeconds), cancellationTokenSource.Token);
+                        }
```
Comment (tongwu-msft) [inline src/Microsoft.Health.TaskManagement/JobHosting.cs:76]: For worker threads, we need to make sure no exception for all operations, I believe redundant try catch is necessary.

Comment (SergeyGaluzo) [reply src/Microsoft.Health.TaskManagement/JobHosting.cs:76]: Are you talking about try/catch around ExecuteJobAsync in ExecuteAsync method?
If so, then ExecuteJobAsync is already protected by try/catch logic. I would understand the need for redundant try/catch in ExecuteAsync if ExecuteJobAsync was a method from a different class. But ExecuteJobAsync method is private, so its sole purpose is to be right for JobHosting class. 
Adding redundant try/catch in this particular case does not make sense to me.

Comment (tongwu-msft) [reply src/Microsoft.Health.TaskManagement/JobHosting.cs:76]: I mean there could be some other operations like _queueClient.IsInitialized() and possible further more.

Comment (SergeyGaluzo) [reply src/Microsoft.Health.TaskManagement/JobHosting.cs:76]: @tongwu-msft Did these problems exist in the original code (before PR)?

Comment (tongwu-msft) [reply src/Microsoft.Health.TaskManagement/JobHosting.cs:76]: Previous if there's any exception throw, the JobHosting would stop and the process for PaaS would be able to restart to recover, if we wait for all workers complete then my concern if any exception in 1 worker would only stop this thread and would not recover.

**Review by tongwu-msft — APPROVED** [review]

**Review by PTaladay — APPROVED** [review]

**Pull request merged**